autoload/lsp.vim

```diff
@@ -166,7 +166,7 @@ function! s:on_text_document_did_open() abort
     let l:buf = bufnr('%')
     call lsp#log('s:on_text_document_did_open()', l:buf, &filetype, getcwd(), lsp#utils#get_buffer_uri(l:buf))
     for l:server_name in lsp#get_whitelisted_servers()
-        call s:ensure_flush(l:buf, l:server_name, function('s:Noop'))
+        call s:ensure_flush(l:buf, l:server_name, {result->s:update_file_content(l:buf, l:server_name, s:get_text_document_text(l:buf))})
```

Comment:
Curious if you had any particular reason why we don't call `s:update_file_content` in `s:ensure_open` to be consistent with `s:ensure_changed`. Is it because you want to make sure it has been opened by the server so in case open fails it can retry?

Reply:
Also seems like we have removed. `s:requires_eol_at_eof`. More discussion about it can be found at https://github.com/prabirshrestha/vim-lsp/pull/158. It actually breaks pyls.

Reply:
I'm not thinking LSP must use real line-ending since document is possibly encoded UTF-16 that contains 0x00. JSON-RPC should always be UTF-8 encoded.